### ISSUE TITLE
Issue installing tritonclient on macos M2

### ISSUE THREAD

**Issue by casassg** [triton-inference-server/server#6246]
**Description**
Latest release from tritonclient in PyPi (https://pypi.org/project/tritonclient/2.37.0.9383150/) seems to include a change which makes cuda-python required dependency for tritonclient (https://github.com/triton-inference-server/client/blob/main/src/python/library/requirements/requirements.txt#L27) even though it's only used for (https://github.com/triton-inference-server/client/blob/d257c0e5c3de6e15d6ef289ff2b96cecd0a69b5f/src/python/library/tritonclient/utils/cuda_shared_memory/__init__.py#L34)

Seems cuda-python is not available for Macbooks which makes tritonclient 2.37.0 not installable on macos.

**Triton Information**
https://pypi.org/project/tritonclient/2.37.0.9383150/

**To Reproduce**

From an M1/2 Macbook with Python 3.11 run:
- `pip install --dry-run tritonclient==2.37.0.9383150`

```
ERROR: Could not find a version that satisfies the requirement cuda-python (from tritonclient) (from versions: none)
ERROR: No matching distribution found for cuda-python
```

**Expected behavior**

Package is installable.

**Comment by kthui** [comment]
@tanmayv25 @jbkyang-nvi Is the dependency on cuda-python expected?

**Comment by tanmayv25** [comment]
As explained in the issue the dependency has been added with this commit: https://github.com/triton-inference-server/client/commit/f880c62996facc9d3e8de954a033b008fd68e48f
Adding optional dependency for cuda shared memory in tritonclient package can allow the package to be installed on systems that does not support cuda-python.

**Comment by yeahdongcn** [comment]
Using container or virtual machine on M2 mac machine would be easier.

**Comment by casassg** [comment]
@yeahdongcn that  seems highly incovenient for simple use cases which just need to access the library for the triton grpc client. Specially for small use cases where you just want to run a small inference load against the server (which is already on a container).

Adding this dependency as required for an optional utility which already has a try/except to check if dep is installed seems  rather inconvenient to users which develop from a Macbook

**Comment by yeahdongcn** [comment]
Yeah, that doesn't straightforward like `pip install ...`. It could be considered as a workaround before everything works on mac.

I'm using https://orbstack.dev/ for setting up VM, it enables the terminal integration. The experience is more seamless from my point of view.

**Comment by casassg** [comment]
https://github.com/triton-inference-server/client/pull/394

**Comment by shshe** [comment]
We're also running into this issue with the latest 2.37 package on M-series Macs. We're looking to use tritonclient as a client while the inference servers are running on our Kubernetes cluster. Requiring a CUDA dependency seems unnecessary for using just the client portion.

**Comment by rusmux** [comment]
Also cannot install tritonclient==2.38.0 on Mac M1

**Comment by tanmayv25** [comment]
The issue should be fixed starting from 23.11(v2.40).

**Comment by churmuzache** [comment]
Did this issue get fixed? I am still getting the same error when I use the triton client 2.39.0

**Comment by jbkyang-nvi** [comment]
> Did this issue get fixed? I am still getting the same error when I use the triton client 2.39.0

`r23.11` is not out yet. It will be available in Triton `2.40.0` available end of this month (November)

**Comment by lucaboulard** [comment]
Same issue with version `tritonclient==2.46.0. Is that normal?
